src/components/CreateAutomationModal.tsx

```diff
@@ -1,19 +1,47 @@
-import { useContext } from "react"
-import { ModalContext } from "~/state/ModalContext"
+import { type MouseEvent, useContext } from "react"
+import { z } from "zod";
+import { ModalContext, type ModalContextType } from "~/state/ModalContext"
 
-const CreateAutomationModal = () => {
+const eventTypeSchema = z.array(z.object({
+  id: z.number(),
+  name: z.enum(['Calendar', 'Mail', 'GitHub', '']),
+  description: z.string()
+}))
+
+type eventTypes = z.infer<typeof eventTypeSchema>
 
-  const { setCreateModalOpened, setRulesetModalOpened } = useContext(ModalContext)
+const CreateAutomationModal = () => {
+  
+  const { setCreateModalOpened, setRulesetModalOpened, setEventTrigger } = useContext(ModalContext)
+  
+  const eventTypes: eventTypes = [
+    { id: 1, name: 'Mail', description: 'Example: When an email was receieved...' },
+    { id: 2, name: 'GitHub', description: 'Example: When a PR has been merged to a repository...' },
```

Comment:
if you use these Mail/Github/Calendar strings all over then you might want to make some enum for it

Reply:
What do you mean exactly? The name in the eventTypes is an enum from the eventTypeSchema.

Reply:
Instead of hardcoding the string everywhere you need it, how about something like https://www.typescriptlang.org/docs/handbook/enums.html#string-enums ?

then if you want to change the string you just need to update the enum definition